File: public/static/templates/spring-boot-clean/src/main/java/com/eventstorming/external/ExternalService.java
forEach: RelationCommandInfo
fileName: {{commandValue.aggregate.namePascalCase}}Service.java
path: {{boundedContext.name}}/{{{options.packagePath}}}/external
---
package {{options.package}}.external;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import java.util.Date;

{{#if boundedContext.fallback}}
@FeignClient(name="{{commandValue.boundedContext.name}}", url="http://{{commandValue.boundedContext.name}}:8080", fallback={{commandValue.aggregate.namePascalCase}}ServiceImpl.class)
{{else}}
@FeignClient(name="{{commandValue.boundedContext.name}}", url="http://{{commandValue.boundedContext.name}}:8080")
{{/if}}
public interface {{commandValue.aggregate.namePascalCase}}Service {
{{#if commandValue.restRepositoryInfo.getMethod}}
    @RequestMapping(method= RequestMethod.GET, path="/{{commandValue.aggregate.namePlural}}/{{wrap commandValue.aggregate.keyFieldDescriptor.name}}")
    public {{commandValue.aggregate.namePascalCase}} get{{commandValue.aggregate.namePascalCase}}(@PathVariable("{{commandValue.aggregate.keyFieldDescriptor.name}}") {{commandValue.aggregate.keyFieldDescriptor.className}} {{commandValue.aggregate.keyFieldDescriptor.name}});
{{else if commandValue.restRepositoryInfo.method}}
    @RequestMapping(method= RequestMethod.{{commandValue.restRepositoryInfo.method}}, path="/{{commandValue.aggregate.namePlural}}")
    public void {{commandValue.nameCamelCase}}(@RequestBody {{commandValue.aggregate.namePascalCase}} {{commandValue.aggregate.nameCamelCase}});
{{else}}
    @RequestMapping(method= RequestMethod.GET, path="/{{commandValue.aggregate.namePlural}}/{{wrap commandValue.aggregate.keyFieldDescriptor.name}}")
    public {{commandValue.aggregate.namePascalCase}} get{{commandValue.aggregate.namePascalCase}}(@PathVariable("{{commandValue.aggregate.keyFieldDescriptor.name}}") {{commandValue.aggregate.keyFieldDescriptor.className}} {{commandValue.aggregate.keyFieldDescriptor.name}});
{{/if}}

}


<function>

  window.$HandleBars.registerHelper('log', function (aggregate) {
      console.log("template log:", aggregate)
    return aggregate;
  })

  window.$HandleBars.registerHelper('wrap', function (exp) {
    return '{'+exp+'}';
  })

</function>

File: public/static/templates/nodejs/external/ExternalService.js
forEach: RelationCommandInfo
fileName: {{commandValue.aggregate.namePascalCase}}Service.js
path: {{boundedContext.name}}/external
---
const request = require('request');

{{#MethodGet commandValue.restRepositoryInfo.method}}
function {{commandValue.namePascalCase}}(callback){
    var options = {
        uri : 'http://localhost:8080/{{commandValue.boundedContext.namePlural}}',
        json : true
    };

    request.get(options, function(error, response, body){
        return callback(body);
    });
}
{{/MethodGet}}

{{#MethodPost commandValue.restRepositoryInfo.method}}
function {{commandValue.namePascalCase}}(data, callback){
    var options = {
        uri : 'http://localhost:8080/{{commandValue.boundedContext.namePlural}}',
        body : data,
        json : true
    };

    request.post(options, function(error, response, body){
        return callback(body);
    })
}
{{/MethodPost}}

{{#MethodUpdate commandValue.restRepositoryInfo.method}}
function {{commandValue.namePascalCase}}(id, data, callback){
    var options = {
        uri : 'http://localhost:8080/{{commandValue.boundedContext.namePlural}}/'+id,
        body : data,
        json : true
    };

    request.put(options, function(error, response, body){
        return callback(body);
    })
}
{{/MethodUpdate}}

{{#MethodDelete commandValue.restRepositoryInfo.method}}
function {{commandValue.namePascalCase}}(id, callback){
    var options = {
        uri : 'http://localhost:8080/{{commandValue.boundedContext.namePlural}}/'+id,
        body : data,
        json : true
    };

    request.delete(options, functino(error, response, body){
        return callback(body);
    })
}
{{/MethodDelete}}

module.exports = {{commandValue.namePascalCase}}

<function>
	window.$HandleBars.registerHelper('MethodGet', function(method, options){
        if(method.endsWith('GET')){
        	return options.fn(this)
		}
		else{
			return options.inverse(this)
		}
    });
	window.$HandleBars.registerHelper('MethodPost', function(method, options){
        if(method.endsWith('POST')){
        	return options.fn(this)
		}
		else{
			return options.inverse(this)
		}
    });
	window.$HandleBars.registerHelper('MethodUpdate', function(method, options){
        if(method.endsWith('PUT')){
        	return options.fn(this)
		}
		else{
			return options.inverse(this)
		}
    });
	window.$HandleBars.registerHelper('MethodDelete', function(method, options){
        if(method.endsWith('DELETE')){
        	return options.fn(this)
		}
		else{
			return options.inverse(this)
		}
    });
</function>